test/support/spec.rb

```diff
@@ -1335,13 +1326,31 @@ def run_specs
         formatter.print_error(*@errors.last)
       else
         formatter.print_success(context, test)
+      ensure
```

Comment:
Adjusting where `after(:each)` and `after(:all)` live in the spec runner.  Previously there were problems where exceptions might prevent the after clause from running, and the after(:all) was being run after the first test, which was not good.

This is "more correct" but still not 100% correct, as all of the `after(:all)` clauses are run completely at the end of the test rather than executed on the final spec of each context